lib/QubitLftSyncer.class.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+class QubitLftSyncer
+{
+    private $parentId;
+    private $limit;
+
+    public function __construct($parentId, $limit = 10000)
+    {
+        $this->parentId = $parentId;
+        $this->limit = $limit;
+    }
+
+    public function sync()
+    {
+        // Get checksum representing lft values of DB and Elasticsearch
+        $dbChecksum = $this->getChildLftChecksumForDB();
+        $esChecksum = $this->getChildLftChecksumForElasticsearch();
+
+        // If checksums don't match then repair
+        if ($dbChecksum != $esChecksum) {
+            $this->repairEsChildrenLftValues();
+
+            // Check to see if repair worked
+            $esChecksum = $this->getChildLftChecksumForElasticsearch();
+
+            if ($esChecksum != $dbChecksum) {
+                // Try to wait until Elasticsearch repair completes
+                $waitCount = 0;
+                $maxWaitAttempts = 3;
+
+                do {
+                    sleep(1);
+                    ++$waitCount;
+                    $esChecksum = $this->getChildLftChecksumForElasticsearch();
+                } while ($waitCount < $maxWaitAttempts && $esChecksum != $dbChecksum);
+            }
+
+            return true;
```

Comment:
This will return true even if the repair could not be verified after the retries.